### PR TITLE
[CAT-515] FIX: force decompress when deserializing if url ends in gz

### PR DESCRIPTION
**Pull request by meghanhickey**
In 5.x, the storage object returned by Document Extraction has a content type of `application/x-gzip`, but in 6.0 the content type is `binary/octet-stream`.

Since we can't always tell by the content type if the response needs to be decompressed, check the request URL and decompress the response if need be.

### CODE REVIEW

**Review by fitzworkhub — COMMENTED** [review]
Do we know why it's unclear why some need to be compressed and others don't?

**Comment by meghanhickey** [comment]
> Do we know why it's unclear why some need to be compressed and others don't?

The issue is that the content type header for gzip files changed from 5.x to 6.x, so we can no longer rely on that header alone to tell if we need to decompress the file. I filed a bug on the PS board for the header change: https://indicodata.atlassian.net/browse/DEV-10316

File: indico/http/client.py [hunk 1]
```diff
@@ -155,6 +155,10 @@ def _make_request(
         if len(url_parts) > 1 and (url_parts[-1] == "json" or url_parts[-2] == "json"):
             json = True
 
+        decompress = False
+        if len(url_parts) > 1 and (url_parts[-1] == "gz"):
+            decompress = True
+
```
Comment (sihrc) [inline indico/http/client.py:161]: decompress = len(url_parts) > 1 and (url_parts[-1] == "gz")

path.split is typically better handled by pathlib to get an extension
`decompress: bool = Path(path).suffix == "gz"`

File: indico/http/serialization.py [hunk 2]
```diff
@@ -48,6 +44,10 @@ def deserialize(response, force_json=False):
         )
 
 
+
+def raw_bytes(content, *args, **kwargs):
+    return content
```
Comment (sihrc) [inline indico/http/serialization.py:49]: Where is this used?

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
 def pytest_addoption(parser):
     parser.addoption(
-        "--host", action="store", default="dev.indico.io", help="indico ipa host"
+        "--host", action="store", default="dev-ci.us-east-2.indico-dev.indico.io", help="indico ipa host"
```
Comment (sihrc) [inline tests/integration/conftest.py:12]: In terms of backwards compatibility, DEVOPS should just setup dev.indico.io CNAME to dev-ci.us-east-2.indico-dev.indico.io

File: indico/http/serialization.py (hunk 2 above)
Comment (meghanhickey) [reply indico/http/serialization.py:49]: https://github.com/IndicoDataSolutions/indico-client-python/pull/213/files#diff-ef5fa222e2bc2fb27c766b726185367b39ec9ef84ca3ac887ebec44f0b2dc7fcR74

File: tests/integration/conftest.py (hunk 3 above)
Comment (sihrc) [reply tests/integration/conftest.py:10]: @meghanhickey sorry :( I think there's also an issue with the SSL cert not accepting dev.indico.io after all. We might just need to change these.

**Review by goatrocks — APPROVED** [review]
are there any changes we should make to 6.x to have the right content type come back for users who may not update their SDKs as quickly?

**Pull request merged**